azure-pipelines.yml

```diff
@@ -56,41 +57,44 @@ jobs:
-          vcpkg install boost-filesystem:x64-windows boost-test:x64-windows boost-program-options:x64-windows tbb:x64-windows eigen3:x64-windows cgal:x64-windows
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          choco install -y ninja --force --force-dependencies
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-        displayName: "Install build dependencies"
+        git submodule update --init
+        python -m pip install --user -r ext/gudhi-deploy/build-requirements.txt
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        python -m pip install --user -r ext/gudhi-deploy/test-requirements.txt
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        # Only vcpkg release libs for CI
+        echo.set(VCPKG_BUILD_TYPE release)>> C:\vcpkg\triplets\x64-windows.cmake
+        vcpkg install boost-filesystem:x64-windows boost-test:x64-windows boost-program-options:x64-windows tbb:x64-windows eigen3:x64-windows cgal:x64-windows
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        choco install -y ninja --force --force-dependencies
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        displayName: 'Install build dependencies'
       - script: |
-          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" amd64
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release -G "Ninja" -DFORCE_EIGEN_DEFAULT_DENSE_INDEX_TYPE_TO_INT=ON $(cmakeVcpkgFlags) $(cmakeFlags) ..
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          ninja
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          ctest --output-on-failure -C Release -E diff_files
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          cmake  -DWITH_GUDHI_PYTHON=ON -DWITH_GUDHI_REMOTE_TEST=ON .
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          cd src\python
-          copy "C:\vcpkg\installed\x64-windows\bin\mpfr*.dll" ".\gudhi\"
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          copy "C:\vcpkg\installed\x64-windows\bin\gmp*.dll" ".\gudhi\"
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          copy "C:\vcpkg\installed\x64-windows\bin\tbb*.dll" ".\gudhi\"
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          python setup.py build_ext --inplace
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-          SET PYTHONPATH=%CD%;%PYTHONPATH%
-          echo %PYTHONPATH%
-          ctest --output-on-failure -C Release
-          IF %ERRORLEVEL% NEQ 0 EXIT 1
-        displayName: "Build and test"
+        $ErrorActionPreference = 'Stop'
+        call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+        mkdir build
+        cd build
+        cmake -DCMAKE_BUILD_TYPE=Release -G "Ninja" -DFORCE_EIGEN_DEFAULT_DENSE_INDEX_TYPE_TO_INT=ON $(cmakeVcpkgFlags) $(cmakeFlags) ..
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        ninja
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        ctest --output-on-failure -C Release -E diff_files
+        IF %ERRORLEVEL% NEQ 0 EXIT 1
+        for /f %%i in ('where python') do set PYTHON_EXE=%%i
```

Comment:
Er, this sets PYTHON_EXE to the *last* entry returned by `where python`? That's a bit strange.

Reply:
Because it seems to be the _good way_ to do it in batch :sweat_smile: cf. https://devblogs.microsoft.com/oldnewthing/20120731-00/?p=7003

If you prefer, I can do:
```batch
          for /f %%i in ('where python') do (set PYTHON_EXE=%%i & goto :next)
          :next
```

Reply:
I guess my question is not so much the syntax (this is CI, I don't much care), it is more trying to understand why we need to specify python now when we didn't before. What versions of Python exist under what names on that machine, which one do we want (apparently the only one called plain `python`?), and which one does CMake choose (and ideally why). Again, I don't really care about the CI case in particular, I care more that this is a fallout of the cmake change which could also affect users.

Reply:
You are right, and I did not talk too much about my experiments around this new feature, my fault too.
If you have a look to https://github.com/VincentRouvreau/play_with_azure_pipelines/blob/159a010d99352edd169949c700af23940b5eabe0/azure-pipelines.yml#L44-L58

In a CMakeLists.txt, I just call `find_package( Python COMPONENTS Interpreter)`.

1. I first call `cmake` without any arguments, I get on [Azure CI](https://dev.azure.com/vincentrouvreau/play_with_azure_pipelines/_build/results?buildId=1171&view=logs&jobId=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=3d94a178-cdfb-5c90-a971-5deeccbbe043) (you may not have access to the logs):
`-- Found Python: C:/hostedtoolcache/windows/Python/3.12.4/x64/python3.exe (found version "3.12.4") found components: Interpreter
`
Note this is not correct, because I said in the yaml file:
```yml
    steps:
      - task: UsePythonVersion@0
        displayName: Use Python $(pythonVersion)
        inputs:
          versionSpec: 3.9
          addToPath: true
          architecture: "x64"
```

2. Then I call:
```batch
for /f %%i in ('where python') do echo %%i
# Prints: C:\hostedtoolcache\windows\Python\3.9.13\x64\python.exe
for /f %%i in ('where python') do (set PYTHON_EXE=%%i & goto :next)
:next
cmake -DPython_EXECUTABLE=%PYTHON_EXE% ..
# -- Found Python: C:\hostedtoolcache\windows\Python\3.9.13\x64\python.exe (found version "3.9.13") found components: Interpreter
```

I don't know how it will affect the users, but for sure it won't affect main users that are only using conda or pip packages.
I also don't know how the python installation is performed on the Azure Windows virtual machines, but the behaviour is quite strange... I can have a look if there is something that could replace `UsePythonVersion` or if its installation is not well performed.

Reply:
So the system has other versions of Python installed in addition to the one we request, that's fine. The one we request is the only one in the default search path, which is good. However, Python_FIND_REGISTRY is FIRST by default, meaning "Try to use registry before environment variables.", and the registry probably knows both versions of Python and not our preferences. It looks like we could pass `-DPython_FIND_REGISTRY=LAST` (or NEVER) to cmake in this CI. Does that work?

Reply:
I used `-DPython_FIND_REGISTRY=LAST` and `-DPython_FIND_FRAMEWORK=LAST` on [d190e7d](https://github.com/GUDHI/gudhi-devel/pull/1112/commits/d190e7d2a3bff37608cc6da429d570d61b69004c)